### PR TITLE
Building friend relation

### PR DESCRIPTION
**Pull request by sangsoora**
Friendship function is live now. For the moment you can send friend request on trip show page and you can accept or decline received request, check sent request and see the friends list on friend requests page I created. Link is on the navbar. For more details, please check commits.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:first_name, :last_name, :email, :home_city, :latitude, :longitude])
+
+    devise_parameter_sanitizer.permit(:account_update, keys: [:first_name, :last_name, :email, :home_city, :phone_number, :latitude, :longitude])
```
Comment (jiggyjensen) [inline app/controllers/application_controller.rb:9]: what is this method doing?

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class FriendshipsController < ApplicationController
+  before_action :set_friendship, only: [:destroy]
+
+  def create
+    @friendship = Friendship.new
+    request = FriendRequest.find(params[:friend_request_id])
```
Comment (jiggyjensen) [inline app/controllers/friendships_controller.rb:6]: why does request not need to be an instance variable here?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (sangsoora) [reply app/controllers/application_controller.rb:9]: It's like strong params for other controller but for users with devise, you need to do this to add more fields to sign up and login

Comment (jiggyjensen) [reply app/controllers/application_controller.rb:9]: oh gotcha- are we asking users to input their lat and long? or do those two fields just have to be included here to be generated once home city is inputted?

File: app/controllers/friendships_controller.rb (hunk 2 above)
Comment (sangsoora) [reply app/controllers/friendships_controller.rb:6]: What I'm doing here is when you accept a friend request, we are creating new empty Friendship and in the params we have friend request id so we are finding a friend request with params friend request id then updating the friendship

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (sangsoora) [reply app/controllers/application_controller.rb:9]: Lat and long will be auto generated by geocoder or google maps api but need to be there to put in the database when users input home city

**Pull request merged**